pkg/apis/core/v1alpha1/logicalcluster_types.go

```diff
@@ -0,0 +1,175 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+)
+
+// LogicalCluster describes the current workspace.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=`.metadata.labels['tenancy\.kcp\.dev/phase']`,description="The current phase (e.g. Scheduling, Initializing, Ready, Deleting)"
+// +kubebuilder:printcolumn:name="URL",type=string,JSONPath=`.status.URL`,description="URL to access the workspace"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+type LogicalCluster struct {
+	v1.TypeMeta `json:",inline"`
+	// +optional
+	v1.ObjectMeta `json:"metadata,omitempty"`
+	// +optional
+	Spec LogicalClusterSpec `json:"spec,omitempty"`
+	// +optional
+	Status LogicalClusterStatus `json:"status,omitempty"`
+}
+
+const (
+	// LogicalClusterName is the name of the LogicalCluster singleton.
+	LogicalClusterName = "this"
+
+	// LogicalClusterFinalizer attached to new ClusterWorkspace (in phase LogicalClusterPhaseScheduling) resources so that we can control
+	// deletion of LogicalCluster resources
+	LogicalClusterFinalizer = "core.kcp.dev/logicalcluster"
+
+	// LogicalClusterTypeAnnotationKey is the annotation key used to indicate
+	// the type of the workspace on the LogicalCluster object. Its format is "root:ws:name".
+	LogicalClusterTypeAnnotationKey = "internal.tenancy.kcp.dev/type"
+)
+
+// LogicalClusterPhaseType is the type of the current phase of the logical cluster.
+//
+// +kubebuilder:validation:Enum=Scheduling;Initializing;Ready
+type LogicalClusterPhaseType string
+
+const (
+	LogicalClusterPhaseScheduling   LogicalClusterPhaseType = "Scheduling"
+	LogicalClusterPhaseInitializing LogicalClusterPhaseType = "Initializing"
+	LogicalClusterPhaseReady        LogicalClusterPhaseType = "Ready"
+)
+
+// LogicalClusterInitializer is a unique string corresponding to a logical cluster
+// initialization controller for the given type of workspaces.
+//
+// +kubebuilder:validation:Pattern:="^([a-z0-9]([-a-z0-9]*[a-z0-9])?(:[a-z0-9]([-a-z0-9]*[a-z0-9])?)*(:[a-z0-9][a-z0-9]([-a-z0-9]*[a-z0-9])?))|(system:.+)$"
+type LogicalClusterInitializer string
+
+// LogicalClusterSpec is the specification of the LogicalCluster resource.
+type LogicalClusterSpec struct {
+	// DirectlyDeletable indicates that this workspace can be directly deleted by the user
+	// from within the workspace.
+	//
+	// +optional
+	// +kubebuilder:default=false
```

Comment:
Global comment: so the godoc renders correctly and does not include the marker tags, the comments should be structured as follows:

```go
// +markersGoHere
// +moreMarkers

// DirectlyDeletable is a field
DirectlyDeletable bool
```

Reply:
just following the convention that is widely established. If we want to change that, let's do it everywhere (outside of this PR) and document it.